### PR TITLE
Added file detail component

### LINKED ISSUE #14: New Component/Route for File detail
We want to see the detail of a File object

with the all file details, and we want to get a download link on demand

Notes:
 - file download links are expiring links to the resource in s3(default is 3600 seconds)
 
## UI Suggestions

 File detail view shows all the file attributes:
  - file_name (can be very long, so maybe shorten the name as in the list view)
  - file_size (In bytes
  - md5_digest
 -  Meta list is a {k v} list similar to RupSetTasks Attributes

Also it needs the **GetFile** component (same as file list view)

### query / data example (from PROD)
```
query one_file {
  node(id: "RmlsZToz") {
    ... on File {
      file_name
      file_size
      md5_digest
      meta {
        k
        v
      }
    }
  }
}

```

```
{
  "data": {
    "node": {
      "file_name": "RupSet_Az_FM(CFM_0_3_SANSTVZ)_mxSbScLn(0.5)_mxAzCh(60.0)_mxCmAzCh(560.0)_mxJpDs(5.0)_mxTtAzCh(60.0)_thFc(0.0).zip",
      "file_size": 29607569,
      "md5_digest": "XvSOseQFqYd7UR+GSv5ijw==",
      "meta": [
        {
          "k": "max_sections",
          "v": "2000"
        },
        {
          "k": "down_dip_width",
          "v": "0.5"
        },
        {
          "k": "connection_strategy",
          "v": "UCERF3"
        },
        {
          "k": "crustal_filename",
          "v": "None"
        },
        {
          "k": "filekey",
          "v": "None"
        },
        {
          "k": "fault_model",
          "v": "CFM_0_3_SANSTVZ"
        },
        {
          "k": "max_jump_distance",
          "v": "5.0"
        },
        {
          "k": "max_cumulative_azimuth",
          "v": "560.0"
        },
        {
          "k": "min_sub_sects_per_parent",
          "v": "2"
        },
        {
          "k": "thinning_factor",
          "v": "0.0"
        },
        {
          "k": "scaling_relationship",
          "v": "TMG_CRU_2017"
        }
      ]
    }
  }
}
```

### PR DESCRIPTION
**Pull request by wooneh**
Closes #14 

File detail component with download link and meta table

image

### CODE REVIEW

File: src/components/FileDetail.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      <Typography variant="h5" gutterBottom>
+        File Detail: Id Not Found
+      </Typography>
+    );
+  }
+
+  return (
+    <>
+      <Typography variant="h5" gutterBottom>
+        File Detail (id: {data?.node?.id})
```
Comment (chrisbc) [inline src/components/FileDetail.tsx:47]: @wooneh pls move this field after **MD5 digest**

File: src/components/FileDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      </Typography>
+      <a href={data?.node?.file_url ?? ''}>Download</a>
+      <Typography>
+        <strong>File name:</strong> <TruncateText text={data?.node?.file_name ?? ''} />
```
Comment (chrisbc) [inline src/components/FileDetail.tsx:51]: @wooneh let's remove TruncateText here please

**Review by chrisbc — CHANGES_REQUESTED** [review]
Hi @wooneh this looks good, just two minor tweaks please. 

FYI I'm going to a add a ticket to create navigation between these two views and another to humanise a few things. And I'll figure out how to deploy this (gh pages?)  for some review by wider team :)

**Comment by wooneh** [comment]
> FYI I'm going to a add a ticket to create navigation between these two views and another to humanise a few things. And I'll figure out how to deploy this (gh pages?) for some review by wider team :)

Sounds good! In the past I've had some issues with deploying React apps using GH pages since it wasn't really supported back then (esp. routing), but things might be different now. Let me know if I can help with this :)

File: src/components/FileDetail.tsx (hunk 1 above)
Comment (wooneh) [reply src/components/FileDetail.tsx]: Do you mean like this?

image

Comment (chrisbc) [reply src/components/FileDetail.tsx]: Oh blast - my bad. no I meant the download link, sorry! Next time I won't rely on comments to tag the line :)

Comment (chrisbc) [reply src/components/FileDetail.tsx]: @wooneh I'll commit this PR as is, and we can tweak it later :)

**Review by chrisbc — APPROVED** [review]

**Pull request merged**